src/NetAppFiles/NetAppFiles/Account/GetNetAppFilesAccount.cs

```diff
@@ -63,21 +64,27 @@ public override void ExecuteCmdlet()
                 ResourceGroupName = resourceIdentifier.ResourceGroupName;
                 Name = resourceIdentifier.ResourceName;
             }
-
-            if (Name != null)
-            {
-                var anfAccount = AzureNetAppFilesManagementClient.Accounts.Get(ResourceGroupName, Name);
-                WriteObject(anfAccount.ConvertToPs());
-            }
-            else if (!string.IsNullOrEmpty(this.ResourceGroupName))
+            try
             {
-                var anfAccounts = AzureNetAppFilesManagementClient.Accounts.List(ResourceGroupName).Select(e => e.ConvertToPs());
-                WriteObject(anfAccounts, true);
+                if (Name != null)
+                {
+                    var anfAccount = AzureNetAppFilesManagementClient.Accounts.Get(ResourceGroupName, Name);
+                    WriteObject(anfAccount.ConvertToPs());
+                }
+                else if (!string.IsNullOrEmpty(this.ResourceGroupName))
+                {
+                    var anfAccounts = AzureNetAppFilesManagementClient.Accounts.List(ResourceGroupName).Select(e => e.ConvertToPs());
+                    WriteObject(anfAccounts, true);
+                }
+                else
+                {
+                        var anfAccounts = AzureNetAppFilesManagementClient.Accounts.ListBySubscription().Select(e => e.ConvertToPs());
+                        WriteObject(anfAccounts, true);
+                }
             }
-            else
+            catch (ErrorResponseException ex)
             {
-                var anfAccounts = AzureNetAppFilesManagementClient.Accounts.ListBySubscription().Select(e => e.ConvertToPs());
-                WriteObject(anfAccounts, true);
+                throw new ErrorResponseException(ex.Body.Error.Message, ex);
```

Comment:
We suggest to use `AzPSCloudException`, this exception can be telemtried.

Reply:
Curiously the Generagted code changed to `ErrorResponseException` when we reference from common types,  updating